classes/migration/upgrade/v3_4_0/I9039_DropDeprecatedFields.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+/**
+ * @file classes/migration/upgrade/v3_4_0/I9039_DropDeprecatedFields.php
+ *
+ * Copyright (c) 2023 Simon Fraser University
+ * Copyright (c) 2023 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class I9039_DropDeprecatedFields
+ *
+ * @brief Drop deprecated fields
+ */
+
+namespace APP\migration\upgrade\v3_4_0;
+
+use Exception;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class I9039_DropDeprecatedFields extends \PKP\migration\upgrade\v3_4_0\I9039_DropDeprecatedFields
+{
+    /**
+     * Run the migration.
+     */
+    public function up(): void
+    {
+        parent::up();
+
+        // Release the index assigned to the column before dropping it
+        try {
+            Schema::table('publication_formats', fn (Blueprint $table) => $table->dropIndex('publication_format_submission_id'));
+        } catch (Exception $e) {
+            $this->_installer->log("Failed to remove the index publication_format_submission_id from the entity publication_formats, we'll assume it doesn't exist and proceed");
```

Comment:
Rather than warning on an error if the index doesn't exist or couldn't be dropped, I'd suggest checking existence explicitly before trying to drop it; we already have doctrine-dbal as a Composer dependency, so can follow this existing pattern: https://github.com/pkp/pkp-lib/blob/07e5a76cd373ced748d12142044ef537e0baaf90/classes/migration/upgrade/v3_4_0/I6093_AddForeignKeys.php#L210

Reply:
Yep! I wanted to, but I didn't want to add an extra package just for that. As the dbal is a direct dependency, I'll move to it 👌
There are other places to apply the same idea.